ietf/ietfauth/forms.py

```diff
@@ -20,8 +21,21 @@
 from .widgets import PasswordStrengthInput, PasswordConfirmationInput
 
 
+validate_username_email = ASCIIUsernameValidator(
+    message=(
+        "This value may contain only unaccented lowercase a-z "
+        "and uppercase A-Z letters, numbers, and @/./+/-/_ characters."
+    )
+)
+
+
+class UsernameEmailField(forms.EmailField):
+    """Email that is suitable for a username"""
+    default_validators = forms.EmailField.default_validators + [validate_username_email]
+
+
 class RegistrationForm(forms.Form):
-    email = forms.EmailField(label="Your email (lowercase)")
+    email = UsernameEmailField(label="Your email (lowercase)")
```

Comment:
should reconcile (lowercase) here with upper and lowercase in vallidate.

Reply:
will the emailfield validators disallow `foo@bar@baz`? (particularly `foo@@baz`)

Reply:
They implement the checks in [RFC5322](https://datatracker.ietf.org/doc/html/rfc5322#autoid-24) accurately, I believe. I'll double-check.

This change should be giving us all the validation we had before plus the standard Django username validation so it should leave us with a strict subset of the bugs we had before.

I'll update the message thrown by the new username validator and, if it doesn't already give a sensible error for non-lowercase input, I'll validate that as well.

Reply:
Checked - it rejects double-@ (both arrangements you listed) both via in-browser validation and, if you bypass that, in Django's validation. It also already complains if you use upper-case letters so it's just the error message on the new validator that needs to change.

Reply:
Updated the message and refactored the case validation so it also gets enforced when selecting a username from active addresses.